draft-irtf-cfrg-randomness-improvements.md

```diff
@@ -116,37 +118,39 @@ normative:
 
 Randomness is a crucial ingredient for Transport Layer Security (TLS) and related security protocols.
 Weak or predictable "cryptographically-strong" pseudorandom number generators (CSPRNGs)
-can be abused or exploited for malicious purposes. The Dual_EC_DRBG random number backdoor and Debian bugs
-are relevant examples of this problem.
-An initial entropy source that seeds a CSPRNG might be weak or broken as well, which can also lead to critical and systemic security problems.
-This document describes a way for security protocol participants to augment their CSPRNGs using long-term private keys.
-This improves randomness from broken or otherwise subverted CSPRNGs.
+can be abused or exploited for malicious purposes. An initial entropy source that seeds a
+CSPRNG might be weak or broken as well, which can also lead to critical and systemic security
+problems. This document describes a way for security protocol implementations to augment their
+CSPRNGs using long-term private keys. This improves randomness from broken or otherwise subverted CSPRNGs.
 
 This document is a product of the Crypto Forum Research Group (CFRG) in the IRTF.
 
 --- middle
 
 # Introduction
 
-Randomness is a crucial ingredient for TLS and related transport security protocols.
-TLS in particular uses random number generators (generally speaking, CSPRNGs) to generate several values: session IDs,
+Secure and properly implemented random number generators, or "cryptographically-strong"
+pseudorandom number generators (CSPRNGs),
+should produce output that is indistinguishable from a random string of the same length.
+CSPRNGs are critical building blocks for TLS and related transport security protocols.
+TLS in particular uses CSPRNGs to generate several values: session IDs,
 ephemeral key shares, and ClientHello and ServerHello random values. CSPRNG failures
 such as the Debian bug described in {{DebianBug}} can lead to insecure TLS connections.
 CSPRNGs may also be intentionally weakened to cause harm {{DualEC}}.
 Initial entropy sources can also be weak or broken, and that would lead to insecurity
-of all CSPRNG instances seeded with them.
-In such cases where CSPRNGs are poorly implemented or insecure, an adversary may be
-able to predict its output and recover secret key material used to protect the connection.
+of all CSPRNG instances seeded with them. In such cases where CSPRNGs are poorly
+implemented or insecure, an adversary Adv may be distinguish its output from a random
```

Comment:
"... may be able to distinguish ..." or "may distinguish"

Reply:
Whoops -- fixed!